### ISSUE TITLE
Export to PDF cuts edges too far

### ISSUE THREAD

**Issue by MysticJay** [thundernest/import-export-tools-ng#106]
Hi,
exported PDFs miss a lot of text on the left side.
![grafik](https://user-images.githubusercontent.com/25123137/80971839-9f896980-8e1d-11ea-93c6-ae515f92c2c4.png)

wild guesses:
1) global print settings
2) inch and cm mismatch

**Comment by MysticJay** [comment]
with a Zero left margin most text gets cut.
![grafik](https://user-images.githubusercontent.com/25123137/80972694-b54b5e80-8e1e-11ea-9448-6bda7ea6f159.png)

with a 5 cm margin the text has again a border:
![grafik](https://user-images.githubusercontent.com/25123137/80972556-9056eb80-8e1e-11ea-9030-84a88e0ded65.png)

**Comment by cleidigh** [comment]
@MysticJay 
Can  you let me know what OS/IET versions ?
@cleidigh

**Comment by cleidigh** [comment]
@MysticJay 
I just tried saving the single message to PDF.  The global margins are default (0.5")
that seems to come out okay.  You mentioned metric dimensions.  Does that change
with the locale you are using?

[test-202005041606-T-2956--Suffix1.pdf](https://github.com/thundernest/import-export-tools-ng/files/4577153/test-202005041606-T-2956--Suffix1.pdf)

**Comment by MysticJay** [comment]
Sidenote: 
While trying to reproduce the problem on a different PC I found that when "Message-Archive-Options-TB" (5.1.0) is also loaded export does not open the export dialogue.

**Comment by MysticJay** [comment]
> Can you let me know what OS/IET versions ?

Windows 10.1909 x64 German
TB 68.7.0 (32-Bit)
IET 4.1.0 Problem persists after uninstalling IET, restart TB, reinstall IET.

export from prefs.js:
	Line 60: user_pref("extensions.importexporttools.export.import_warning", false);
	Line 61: user_pref("extensions.importexporttools.printPDF", false);
	Line 62: user_pref("extensions.importexporttools.printPDF.restore_print_silent", true);
	Line 63: user_pref("extensions.importexporttools.printPDF.start", false);
	Line 64: user_pref("extensions.importexporttoolsng.export.format_warning", false);
	Line 65: user_pref("extensions.importexporttoolsng.export.import_warning", false);
	Line 66: user_pref("extensions.importexporttoolsng.printPDF", false);
	Line 67: user_pref("extensions.importexporttoolsng.printPDF.restore_print_silent", true);

**Comment by MysticJay** [comment]
> I just tried saving the single message to PDF (Windows 10, IETng v4.1.0 The global margins are default (0.5")

exactly the settings here.

just did a complete windows update including cumulative. no change. problem persists.

**Comment by cleidigh** [comment]
okay thanks, it's a lot easier for me to do debugging on Windows , I will see what happens when I change locales.  that's really the only difference we have so far.

**Comment by cleidigh** [comment]
@MysticJay 
I am still unable to reproduce, I tried changing the locale , but no change still looks good on my side.
Couple of thoughts:
- have you tried going back to the old version
- are there any other extensions involved
- what happens if you print to PDF from the main menu?
@cleidigh

**Comment by Patuta** [comment]
I'm having the same issue after updating importexport.

**Comment by cleidigh** [comment]
@Patuta 
OS? Versions?
if you updated and saw the problem does that mean that with same system in Thunderbird set up
'But old extension no issue correct?

**Comment by MysticJay** [comment]
Sorry for the delay.
> I am still unable to reproduce, I tried changing the locale , but no change still looks good on my side.

I am unable to repro on a different system either. But today a customer called and complained about 
not being able to print from TB. He has not installed IET. Print Preview does not work and page setup is empty in all fields. Trying to restore the settings also failed. The OS there is Win7 (to be replaced next week). This TB 68.8.0 (32-Bit). Maybe there is a general problem in TB.

> Couple of thoughts:
>     * have you tried going back to the old version
To what Versions?

>     * are there any other extensions involved

I was able to repro on the original system with only IET loaded.
(But remember the other bug I mentioned above https://github.com/thundernest/import-export-tools-ng/issues/106#issuecomment-623896661)

>     * what happens if you print to PDF from the main menu?
I have no "Print to PDF" in the main menue.

M.

**Comment by cleidigh** [comment]
@MysticJay 
It just started to sound like they're some TB print issues.  I took a quick look in Bugzilla and there are quite a number of issues . however nothing I could match to your description.
For your observation, I don't think I understand for sure if you tried rolling back to extension v4.0.1
which does not use the global print parameters.  I am curious if that changes anything. If not, it's a good indication it's an issue with Thunderbird. If the problem goes away then I could disable the new option temporarily until the solution is found.

I also just tried to reproduce the issue with menus/Message Archive Options, and l see that  menus are messed up.  I will create an issue for that.
![image](https://user-images.githubusercontent.com/25272315/81852712-72c60800-9529-11ea-8d09-a5d4a8839eaa.png)

**Comment by Razzmatazzz** [comment]
I'm running latest TB and IET on Windows 10. Tried printing both with no left margin and with the default 0.5" left margin, both worked properly for me. English/US localization.

**Comment by MysticJay** [comment]
>  rolling back to extension v4.0.1

I tried 4.0.4, which is not compatible with TB68.8.0

could you provide a build which logs the creation data/parameters to a file?

**Comment by cleidigh** [comment]
@MysticJay 
yes I think what I will do is to make a test version with a Boolean preference that allows you to use the global options or the original way as well as printing out the relevant parameters.
Since everyone is going to have different install printers I am wondering if that might be an issue

**Comment by cleidigh** [comment]
@MysticJay 
@patuta
@Razzmatazzz  - thanks for reporting your experience

this test version uses an experimental option to choose between the new global preferences or the old style default preferences.
extensions.importexporttoolsng.experimental.printPDF.use_global_preferences => true ( default setting) - change to false for the old-style

I am also attaching example , debug output that shows the settings on my system for both scenarios.
Note the printer for the global preferences option  ===  Microsoft PDF printer

[Global preferences - default experimenta.txt](https://github.com/thundernest/import-export-tools-ng/files/4631425/Global.preferences.-.default.experimenta.txt)


https://github.com/thundernest/import-export-tools-ng/blob/v4.1.1-PDF-test/xpi/import-export-tools-ng-4.1.1-PDF-test1-tb.xpi

**Comment by MysticJay** [comment]
> Note the printer for the global preferences option === Microsoft PDF printer

THIS was the missing hint! 
I opened that printers settings, and saved them again (there is really nothing to configure but the page format "A4"). Now the problem is gone.

from a general point of view the issue could be closed, but I recommend to provide a link to open the settings of the MS-PDF-Printer.

M.

**Comment by MysticJay** [comment]
It was only a hint.
Today the user again used IET and again PDFs were shifted to the left.
Then I used the Microsoft PDF Printer and the same document showed as expected.
Finally I tried again from IET menu and it worked as well.

So something changed after I opened the settings for MS-PDF-Printer.

Going to try the testbuild later

**Comment by cleidigh** [comment]
@MysticJay 
hmm, mixed results
I will be very curious to see the results of using the experimental preference to go back to "default"
settings versus global

**Comment by atscomms** [comment]
I just installed V 4.1.0 into Thunderbird V 68.9.0 32bit on a current Windows 10 PC.
Exporting a single email message results in the left edge missing several characters. However, If I print the same message to Acrobat, the PDF is perfect.

**Comment by cleidigh** [comment]
@atscomms 
I am releasing a new version for 78/68
I am doing a few fixes at the same time, have you tried changing the printer settings
that worked for @MysticJay  ?
Christopher

**Comment by atscomms** [comment]
I am using TB V 78.2.0 and IET V4.1.0, which is incompatible. So I cannot access the add-on

**Comment by swiftjgs** [comment]
I have been having this same problem for a couple of months, where IET export to PDF would leave a few characters at the top and left of the page off the page. I've been just working around it, but I found that MysticJay's hints about the page setup, right at the top of this thread, worked for me. I have...

Windows 10.0.18363 Build 18363
TB 68.12.0 (32-bit)
IET 4.1.0

My margins are the default, 0.5 inches all around. I opened Page Setup and set the top and left to 1.5 inches. After that, IET export to PDF had the new, extremely large, margins at the top and left. I then reset the margins to 0.5 inches all around. IET successfully exports PDF with the familiar (0.5 inch) margins again. I'd be interested to know if this works for anyone else.
